File: xml-data/src/main/java/artofillusion/plugin/LogRecord.java
package artofillusion.plugin;

import com.thoughtworks.xstream.annotations.XStreamAlias;
import com.thoughtworks.xstream.annotations.XStreamAsAttribute;
import com.thoughtworks.xstream.annotations.XStreamConverter;
import com.thoughtworks.xstream.converters.extended.ToAttributedValueConverter;
import lombok.Data;


@Data
@XStreamAlias("log")
@XStreamConverter(value = ToAttributedValueConverter.class, strings = {"text"})
public class LogRecord {
    @XStreamAsAttribute private String version;
    @XStreamAsAttribute private String date;
    @XStreamAsAttribute private String author;
    private String text;
}

File: xml-data/src/main/kotlin/artofillusion/plugin/LogRecord.kt
/* Copyright 2024 by Maksim Khramov

   This program is free software; you can redistribute it and/or modify it under the
   terms of the GNU General Public License as published by the Free Software
   Foundation; either version 2 of the License, or (at your option) any later version.

   This program is distributed in the hope that it will be useful, but WITHOUT ANY
   WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
   PARTICULAR PURPOSE.  See the GNU General Public License for more details. */

package artofillusion.plugin

import com.thoughtworks.xstream.annotations.XStreamAlias
import com.thoughtworks.xstream.annotations.XStreamAsAttribute
import com.thoughtworks.xstream.annotations.XStreamConverter
import com.thoughtworks.xstream.converters.extended.ToAttributedValueConverter

@XStreamAlias("log")
@XStreamConverter(value = ToAttributedValueConverter::class, strings = ["text"])
data class LogRecord(
    @XStreamAsAttribute
    val version: String? = null,
    @XStreamAsAttribute
    private val date: String? = null,
    @XStreamAsAttribute
    val author: String? = null,
    val text: String? = null
)
